scipy/stats/_stats.pyx

```diff
@@ -513,6 +513,94 @@ cdef double _geninvgauss_pdf(double x, void *user_data) nogil except *:
     return math.exp(_geninvgauss_logpdf_kernel(x, p, b))
 
 
+cdef double _phi(double z) nogil:
+    cdef double inv_sqrt_2pi = 0.3989422804014327
+    return inv_sqrt_2pi * math.exp(-0.5 * z * z)
+
+
+cdef double _Phi(double z) nogil:
+    cdef double m_sqrt1_2 = 0.7071067811865475
+    return 0.5 * math.erfc(-z * m_sqrt1_2)
+
+
+cdef double _genstudentized_range_cdf(int n, double[2] x, void *user_data) nogil:
+    # evaluates the integrand of Equation (3) by Batista, et al [2]
+    # destined to be used in a LowLevelCallable
+    q = (<double *>user_data)[0]
+    k = (<double *>user_data)[1]
+    df = (<double *>user_data)[2]
+
+    s = x[1]
+    z = x[0]
+
+    # terms are evaluated within logarithms when possible to avoid overflows
+    log_terms = math.log(k) + (df / 2) * math.log(df) \
+               - (math.lgamma(df / 2) + (df / 2 - 1) * math.log(2)) \
+               + (df - 1) * math.log(s) - (df * s * s / 2) \
+               + math.log(0.3989422804014327) - 0.5 * z * z  # phi estimation.
+
+    # this term is excluded from the logarithm as it results in zero division
+    t4 = math.pow(_Phi(z + q * s) - _Phi(z), k - 1)
```

Comment:
`t4` = "term 4". A leftover from when the other parts of the calc were split into matching `t1`, `t2`, etc. terms before being combined. That terms causes errors to be thrown when put into a log, so I left it as-is. It could probably be moved to the  `return` statement.

Reply:
Let's see what happens above.

Reply:
For now I've moved it into the return statement, pending a final decision above.